auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,376 @@
+        messageTitle = ""
+        messageContent = ""
+        conn_success = False
+        for rec in confs:
+            try:
+                conn_success = True
+                ipHost = rec.sftpip
+                portHost = rec.sftpport
+                usernameLogin = rec.sftpusername
+                passwordLogin = rec.sftppassword
+                # Connect with external server over SFTP, so we know sure that
+                # everything works.
+                srv = pysftp.Connection(host=ipHost, username=usernameLogin,
+                                        password=passwordLogin, port=portHost)
+                srv.close()
+                # We have a success.
+                messageTitle = _("Connection Test Succeeded!")
+                messageContent = _(
+                    "Everything seems properly set up for FTP back-ups!")
+            except Exception as e:
+                conn_success = False
+                messageTitle = _("Connection Test Failed!")
+                if len(rec.sftpip) < 8:
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
+            raise except_orm(
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise UserError(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            if not os.path.isdir(rec.bkp_dir):
+                os.makedirs(rec.bkp_dir)
+            # Create name for dumpfile.
+            bkp_file = '%s_%s.dump.zip' % (
+                time.strftime('%d_%m_%Y_%H_%M_%S'),
+                rec.name)
+            file_path = os.path.join(rec.bkp_dir, bkp_file)
+            fbk = open(file_path, 'wb')
+            db.dump_db(rec.name, fbk)
+            fbk.close()
+            # Check if user wants to write to SFTP or not.
+            if rec.sftpwrite is True:
+                try:
+                    # Store all values in variables
+                    dir = rec.bkp_dir
+                    pathToWriteTo = rec.sftppath
+                    ipHost = rec.sftpip
+                    portHost = rec.sftpport
+                    usernameLogin = rec.sftpusername
+                    passwordLogin = rec.sftppassword
+                    # Connect with external server over SFTP
+                    srv = pysftp.Connection(
+                        host=ipHost,
+                        username=usernameLogin,
+                        password=passwordLogin,
+                        port=portHost)
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
```

Comment:
IMHO if the configuration is wrong, it's better to raise an exception than to auto fix it. It will prevent misconfigurations. You can add this check to an `@onchange` so that the user gets warned before bad things happen.